js/utils/utils.js

```diff
@@ -0,0 +1,116 @@
+const shaLib = require('./sha256.js');
+const ripemd160Lib = require('./ripemd160.js');
+
+module.exports = {
+  serializeHex: (uint8arr) => {
+    if (!uint8arr) {
+      return '';
+    }
+
+    let hexStr = '';
+    for (let i = 0; i < uint8arr.length; i++) {
+      let hex = (uint8arr[i] & 0xff).toString(16);
+      hex = (hex.length === 1) ? '0' + hex : hex;
+      hexStr += hex;
+    }
+
+    return `0x${hexStr.toLowerCase()}`;
+  },
+
+  deserializeHex: (hexStr) => {
+    if (!hexStr) {
+      return new Uint8Array();
+    }
+
+    let hex = ''
+    if (hexStr.slice(0, 2) === '0x') {
+      hex = hexStr.slice(2);
+    } else {
+      hex = hexStr;
+    }
+
+    let a = [];
+    for (let i = 0; i < hex.length; i+=2) {
+      a.push(parseInt(hex.substr(i,2),16));
+    }
+
+    return new Uint8Array(a);
+  },
+
+  sha256: (buf) => {
+    return module.exports.deserializeHex(shaLib(buf));
+  },
+
+  ripemd160: (buf) => {
+    return ripemd160Lib.default(buf);
+  },
+
+  typedArraysAreEqual: (a, b) => {
+    if (a.byteLength !== b.byteLength) return false;
+    return a.every((val, i) => val === b[i]);
+  },
+
+  /// @notice          Converts big-endian bytes to a uint
+  /// @dev             Traverses the byte array and sums the bytes
+  /// @param _b        The big-endian bytes-encoded integer
+  /// @return          The integer representation
+  bytesToUint: (uint8Arr) => {
+    let total = BigInt(0);
+    for (let i = 0; i < uint8Arr.length; i++) {
+      total += BigInt(uint8Arr[i]) << (BigInt(uint8Arr.length - i - 1) * BigInt(8));
+    }
+    return total;
+  },
+
+  safeSlice: (buf, first, last) => {
+    let start;
+    let end;
+
+    if (first === null || undefined) { start = 0; }
+    if (last === null || undefined) { end = buf.length - 1; }
+
+    if (typeof first === 'bigint') {
+      start = Number(first);
```

Comment:
This is unsafe in some cases (for numbers larger than `Number.MAX_SAFE_INTEGER`) but unlikely in practice. I think it would be smarter to throw instead of allowing for unsafe behavior

```
> Number(900719925474099111n)
900719925474099100
```